Event/include/Event/Pulse.h

```diff
@@ -0,0 +1,176 @@
+/**
+ * @file Pulse.h
+ * @brief Class for the input pulses to QIE
+ * @author Niramay Gogate, Texas Tech University
+ */
+
+#ifndef EVENT_PULSE_H
+#define EVENT_PULSE_H
+
+namespace ldmx {
+
+  /**
+   * @class Pulse
+   * @brief The base class to store the most important functions
+   *
+   * @note Always use the classes inherited from this class for 
+   * QIE input pulse.
+   */
+  class Pulse
+  {
+ public:
+
+    /**
+     * Evaluate the pulse at time T
+     */
+    virtual float eval(float T);
+
+    /**
+     * Integrate the pulse from T1 to T2
+     */
+    virtual float Integrate(float T1,float T2);
+
+    /**
+     * Differentiate pulse at time T
+     */
+    virtual float Der(float T);
+
+    /**
+     *  maximum of the pulse
+     */
+    virtual float Max();
+  };
+
+
+  /**
+   * @class Bimoid
+   * @brief Pulse made out of difference of two sigmoids
+   *
+   * @note The pulse maximum is found numerically
+   */
+  class Bimoid: public Pulse
+  {
+ public:
+    /**
+     * Class constructors.
+     */
+    Bimoid(float start,float rise, float fall, float qq);
+    Bimoid(float start,float qq);
+
+    /**
+     * Evaluate the pulse at time T
+     */
+    float eval(float T);
+
+    /**
+     * Integrate the pulse from T1 to T2
+     */
+    float Integrate(float T1,float T2);
+
+    /**
+     *  maximum of the pulse
+     */
+    float Max();
+
+    /**
+     * Differentiate pulse at time T
+     */
+    float Der(float T);
+ private:
+    // starting time of the pulse
```

Comment:
For single line docs of variables, use `///` so it gets picked up when building the documentation.